### PR TITLE
Add Entry - JavaScript: Closures

### PR DESCRIPTION
**Pull request by Dusch4593**
### Description

Created a new entry for JavaScript closures, per our ongoing work with keywords.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/javascript/concepts/closures/closures.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+This example describes the three primary scopes visible to closures:
+
+- `myGlobalString` exists in the global scope since it is not defined inside something else, like a function.
+- From the point-of-view of the `inner()` function, everything outside its scope (outer functions, the global environment, etc.) is in the lexical environment, including `myLexicalString` and `myGlobalString`.
+- Inside the `inner()` is a locally-bound variable, `myLocalString`, that uses all the variables from its lexical environment.
+
+## Codebyte Example
+
+Another common instance of JavaScript closures with with [callbacks](https://www.codecademy.com/resources/docs/javascript/callbacks).
```
Comment (SSwiniarski) [inline content/javascript/concepts/closures/closures.md:46]:
```suggestion
Another common instance of JavaScript closures is with [callbacks](https://www.codecademy.com/resources/docs/javascript/callbacks).
```

File: content/javascript/concepts/closures/closures.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  - 'Algorithms'
+  - 'Conceptual'
+  - 'Dynamic Programming'
+  - 'Functions'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/computer-science'
+---
+
+Closures are [functions](https://www.codecademy.com/resources/docs/javascript/functions) that can refer to [variables](https://www.codecademy.com/resources/docs/javascript/variables) and other bindings beyond its scope, even after being called.
```
Comment (SSwiniarski) [inline content/javascript/concepts/closures/closures.md:17]:
```suggestion
Closures are [functions](https://www.codecademy.com/resources/docs/javascript/functions) that can refer to [variables](https://www.codecademy.com/resources/docs/javascript/variables) and other bindings beyond their scope, even after being called.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Dusch4593, I reviewed this and had a couple of suggestions.

**Comment by Dusch4593** [comment]
@SSwiniarski finished implementing your suggestions 😄

**Review by SSwiniarski — APPROVED** [review]
Looks good for review two. 👍

**Review by yangc95 — COMMENTED** [review]
Just one suggestion. This is really well-worded! Thanks for this entry @Dusch4593 🚀

**Comment by Dusch4593** [comment]
@yangc95 applied your suggestion.

**Review by yangc95 — APPROVED** [review]
LGTM 😃

**Pull request merged**